web/modules/custom/joinup_user/joinup_user.module

```diff
@@ -210,3 +215,75 @@ function joinup_user_form_alter_validate(&$form, FormStateInterface $form_state)
     $active_form->setFormId($build_info['form_id']);
   }
 }
+
+/**
+ * Implements hook_search_api_query_TAG_alter().
+ *
+ * Alters the Search API query for the user content:
+ * - show only the content the user is author of;
+ * - do not show entities of some bundles.
+ */
+function joinup_user_search_api_query_search_api_field_field_user_content_alter(QueryInterface &$query) {
+  /** @var \Drupal\user\Entity\User $user */
+  $user = $query->getOption('search_api_field entity');
+
+  // Do not show contact information and owner entities as they are not
+  // relevant.
+  $query->addCondition('entity_bundle', ['contact_information', 'owner'], 'NOT IN');
```

Comment:
While generally whitelist is indeed better, we don't have any plans to hide any other content here. Actually, this comes from a search, and we index everything that we want to be shown. Only in this case, we don't want to show owners and contact informations as they are shown in the related entity "About" page.